src/RefreshToolbar/RefreshToolbar.tsx

```diff
@@ -0,0 +1,102 @@
+import { FC, ReactNode, useMemo } from 'react';
+import Button from 'src/Button/Button';
+import {
+    ToolbarContainer, LogoLoadingSpinner, RefreshProgressContainer,
+    ToolbarAdditionalContent, ButtonContainer, RightSideContainer
+} from './refreshToolbarStyles';
+import { ReactComponent as RefreshIcon } from './assets/refreshIcon.svg';
+import { ThemeProvider } from '@emotion/react';
+import theme from 'src/styles/theme';
+
+export interface RefreshToolbarProps {
+    /**
+     * This is called when the refresh button is clicked.
+     */
+    handleRefresh(): void;
+    /**
+     * This is to set the Refresh toolbar class.
+     */
+    className?: string;
+    /**
+     * This is to set to false if you want to apply your own container styling (with className prop or a wrapper)
+     */
+    standalone?: boolean;
+    /**
+     * Set this to true if you want to display the loading image.
+     */
+    isRefreshing?: boolean;
+    /**
+     * This is to set the last updated text.
+     */
+    lastUpdated?: string | null;
+    /**
+     * This is to display the loading image label.
+     */
+    refreshProgressLabel?: string;
+    /**
+     * This is to render right side components.
+     */
+    rightComp?: ReactNode;
+    /**
+     * This is to set right side text.
+     */
+    rightSideText?: string;
+    /**
+     * This is to render a custom refresh button. 
+     */
+    refreshButton?: ReactNode;
+}
+
+/**
+ * Refresh toolbar component that can be used as an additional header row.
+ *
+ * Set [standalone] to false if you want to apply your own container styling (with className
+ * prop or a wrapper). Set [standalone] to true to add full-width behavior with margins and
+ * a bottom border (to function as a standalone header). Defaults to true.
+ */
+const RefreshToolbar: FC<RefreshToolbarProps> = props => {
+    const {
```

Comment:
Does the storybook docgen properly document the props when they're destructured outside the function params above? I have a vague memory of not seeing the interface documentation properly inside storybook when I didn't do the destructuring up there.

Reply:
I have checked this and it looks fine in storybook without destructuring the props here. So we are good.